### PR TITLE
Fix quickcheck IR generation, and enable quickcheck tests

### PR DESCRIPTION
**Pull request by andywingo**
* tests/pfquickcheck/pflua_ir.lua: All arithmetic expressions are in the
  uint32 range.  The optimizer might change some of these to be in the
  int32 range, but that's its job.  Ours is to produce semantically
  valid IR that could correspond to valid source pflang expressions.

* tests/Makefile: Now that the htons, NaN, repeatable randomness, and IR
  generation bugs are fixed, enable the quickcheck tests.

* tests/test-filters: Move this file here, out of pfquickcheck/.

### CODE REVIEW

File: tests/pfquickcheck/pflua_ir.lua [hunk 1]
```diff
@@ -15,27 +15,25 @@ function False() return { 'false' } end
 function Fail() return { 'fail' } end
 function ComparisonOp() return choose({ '<', '>' }) end
 function BinaryOp() return choose({ '+', '-', '/' }) end
-function UnaryOp()
-   return choose({ 'uint32', 'int32', 'ntohs', 'ntohl' })
-end
+function UnaryOp() return choose({ 'int32', 'ntohs', 'ntohl' }) end
```
Comment (andywingo) [inline tests/pfquickcheck/pflua_ir.lua]: really the ntohs / ntohl should be around PacketAccess, i think

**Comment by kbara** [comment]
LGTM

**Comment by kbara** [comment]
That looks better. LGTM, though I'm still wondering a bit about the range-analysis output range being int32.

**Comment by andywingo** [comment]
regarding the output range of ntohl, it's fine because ntohl is really bit.bswap(), which produces an int32, and it always appears in source programs wrapped by uint32(), though the uint32() can in many cases be removed: https://github.com/Igalia/pflua/blob/master/src/pf/expand.lua#L1076

**Pull request merged**